src/devices/Tsl256x/Tsl256x.cs

```diff
@@ -0,0 +1,353 @@
+        public Illuminance MeasureAndGetIlluminance()
+        {
+            // We need to make sure we're on otherwise the sensor won't measure anything
+            Enabled = true;
+            // We need to write the timing timer to start a measurement
+            WriteByte(Register.TIMING, (byte)((byte)_integrationTime | (byte)_gain));
+            switch (_integrationTime)
+            {
+                case IntegrationTime.Integration13_7Milliseconds:
+                    Thread.Sleep(14);
+                    break;
+                case IntegrationTime.Integration101Milliseconds:
+                    Thread.Sleep(101);
+                    break;
+                case IntegrationTime.Integration402Milliseconds:
+                    Thread.Sleep(402);
+                    break;
+                case IntegrationTime.Manual:
+                default:
+                    throw new ArgumentOutOfRangeException($"Only non manual integration time are supported");
+            }
+
+            return GetIlluminance();
+        }
+
+        /// <summary>
+        /// Get the calculated Illuminance. Default range is Lux
+        /// </summary>
+        /// <returns>The illuminance</returns>
+        /// <remarks>If you have used the manual integration, you won't be able to use this formula</remarks>
+        public Illuminance GetIlluminance()
+        {
+            // See documentation page 23 and following for the constant and the calculation
+            // Integration time scaling
+            // scale channel values by 2^10
+            const int ChannelScale = 1024;
+            // 322/11 * 2^CH_SCALE
+            const int ChannelScale13_7 = 0x7517;
+            // 322/81 * 2^CH_SCALE
+            const int ChannelScale101 = 0x0fe7;
+
+            GetRawChannels(out ushort ch0, out ushort ch1);
+            if (ch0 == 0)
+            {
+                return Illuminance.FromLux(0);
+            }
+
+            double ratio = ch1 / ch0;
+            // Calculate the integration and scaling
+            double scale = 0;
+            switch (_integrationTime)
+            {
+                case IntegrationTime.Integration13_7Milliseconds:
+                    scale = ChannelScale13_7;
+                    break;
+                case IntegrationTime.Integration101Milliseconds:
+                    scale = ChannelScale101;
+                    break;
+                case IntegrationTime.Integration402Milliseconds:
+                case IntegrationTime.Manual:
+                default:
+                    scale = ChannelScale;
+                    break;
+            }
+
+            scale = _gain == Gain.Normal ? scale : scale * 16;
+
+            double channel0 = (ch0 * scale) / ChannelScale;
+            double channel1 = (ch1 * scale) / ChannelScale;
+
+            double lux = 0;
+            if (_packageType == PackageType.PackageCs)
+            {
+                if (ratio <= 0.52)
```

Comment:
can you point to spec about these calculations?

Reply:
It's done line 233. : `See documentation page 23 and following for the constant and the calculation`